src/reducers/Entities/LocationReducer.js

```diff
@@ -1,11 +1,18 @@
 import { LOCATION_ACTIONS } from '../../actions/Entities';
 import { UIDatabase } from '../../database';
+import { ROUTES } from '../../navigation/index';
+
+const plainLocation = (location = {}) => ({
+  id: location.id,
+  description: location.description,
+  code: location.code,
+});
```

Comment:
This is the weirdest? part of the PR I think.

- Realm objects can't be spread, so if there was a redux object, trying to edit immutably is a pain.
- Using realm `toJSON` results in an object which has clones of all of the related records also, so if you are to use `update` on the resulting object which has been edited, the potentially stale and related `sensor` would also be saved.
- ONLY using realm objects results in live updates and requires more complex rolling back (as we update after a 'save' button is clicked - this is the same as in the new sensor flow).

So, I decided to take this route, and transform the objects into plain, simple JS objects in the redux store.

Another alternative would be to override `toJSON` in the `Sensor` realm class.

side note: Possibility here to create an abstraction to hide realm implementation such that we use plain JS objects by overriding `toJSON` on all of the realm classes and pass it through the abstraction to be persisted as whatever. Good phase1 of 2 for implementing sqlite.

Reply:
Just something like this in a realm class:

```
toJSON(){
	return { this.id ... }
}
```


Bring on sqlite and working with plain js objects 🤷 

renamed and a brief comment: 064e355